src/test_l3_dsDisplay.c

```diff
@@ -0,0 +1,330 @@
+    {"dsDISPLAY_RXSENSE_OFF", (int32_t)dsDISPLAY_RXSENSE_OFF},
+    {"dsDISPLAY_HDCPPROTOCOL_CHANGE", (int32_t)dsDISPLAY_HDCPPROTOCOL_CHANGE},
+    {"dsDISPLAY_EVENT_MAX", (int32_t)dsDISPLAY_EVENT_MAX},
+    {NULL, -1}
+};
+
+// Mapping table for dsVideoAspectRatio_t
+const static stringkey_mapping dsVideoAspect_RatioMappingTable[] = {
+    {"dsVIDEO_ASPECT_RATIO_4x3", (int32_t)dsVIDEO_ASPECT_RATIO_4x3},
+    {"dsVIDEO_ASPECT_RATIO_16x9", (int32_t)dsVIDEO_ASPECT_RATIO_16x9},
+    {"dsVIDEO_ASPECT_RATIO_MAX", (int32_t)dsVIDEO_ASPECT_RATIO_MAX},
+    {NULL, -1}
+};
+
+static char* mapKeyToString(const stringkey_mapping* keyMappingTable, int32_t keyCode) {
+    if (keyMappingTable == NULL) {
+        return NULL;
+    }
+
+    for (int32_t i = 0; keyMappingTable[i].stringValue != NULL; i++) {
+        if (keyMappingTable[i].keyCode == keyCode) {
+            return keyMappingTable[i].stringValue;
+        }
+    }
+
+    return "Unknown value";
+}
+
+/*callback*/
+void DisplayEventCallback(intptr_t handle, dsDisplayEvent_t event, void* eventData)
+{
+    UT_LOG_INFO("DisplayEventCallback dsDisplayEvent_t:[%s] ",mapKeyToString(displayEventMappingTable, event));
+}
+
+void dsDisplay_Init()
+{
+    dsError_t status   = dsERR_NONE;
+
+    UT_LOG_INFO("Calling dsDisplayInit()");
+    status = dsDisplayInit();
+    UT_LOG_INFO("Result dsDisplayInit() dsError_t=[%s]", mapKeyToString(errorMappingTable, status));
+    UT_ASSERT_EQUAL_FATAL(status, dsERR_NONE);
+
+}
+
+void dsDisplay_Term()
+{
+    dsError_t status   = dsERR_NONE;
+
+    UT_LOG_INFO("Calling dsDisplayTerm()");
+    status = dsDisplayTerm();
+    UT_LOG_INFO("Result dsDisplayTerm() dsError_t=[%s]",  mapKeyToString(errorMappingTable, status));
+    UT_ASSERT_EQUAL_FATAL(status, dsERR_NONE);
+}
+
+static void dsGetDisplay_getHandle()
+{
+    dsError_t status   = dsERR_NONE;
+    int32_t choice,port;
+    intptr_t displayHandle = -1;
+
+    dsDisplay_Init();
+
+    UT_LOG_INFO(" \t  Supported Video Port are:");
+    UT_LOG_INFO("------------------------------------------");
+    for (port = 0; port < gDSvideoPort_NumberOfPorts; port++) {
+        UT_LOG_INFO("\t%d.  %-20s\n", port+1, mapKeyToString(videoPortTypeMappingTable, gDSVideoPortConfiguration[port].typeid));
+    }
+    UT_LOG_INFO("------------------------------------------");
+    UT_LOG_INFO(" Select the Video Port:");
+    scanf("%d", &choice);
+    port = choice - 1;
+    UT_LOG_INFO("Port %d choice %d ",port,choice);
+    if(choice < 1 || choice > gDSvideoPort_NumberOfPorts) {
+        UT_LOG_ERROR("Invalid Port choice\n");
+        return;
+    }
+
+    UT_LOG_INFO("Calling dsGetDisplay(type:[%s],index:[%d])",mapKeyToString(videoPortTypeMappingTable, gDSVideoPortConfiguration[port].typeid));
+    UT_LOG_INFO("Typeid: %d ",gDSVideoPortConfiguration[port].typeid);
+    status = dsGetDisplay(gDSVideoPortConfiguration[port].typeid, gDSVideoPortConfiguration[port].index, &displayHandle);
```

Comment:
Please make the dsiplayHandle variable Static global ? else scope of the handle finish with function call and will not be available for subsequent calls.

Reply:
Done